src/main/scala/io/iohk/ethereum/domain/SignedTransaction.scala

```diff
@@ -77,7 +77,10 @@ object SignedTransaction {
 
     for {
       key <- recoveredPublicKey
-      addrBytes = crypto.kec256(key).slice(FirstByteOfAddress, LastByteOfAddress)
+
+      //TODO: Parity doesn't do this (https://github.com/paritytech/parity/blob/master/ethkey/src/keypair.rs#L23-L28)
+      //      more research is needed on why `key.tail` is needed
+      addrBytes = crypto.kec256(key.tail).slice(FirstByteOfAddress, LastByteOfAddress)
```

Comment:
👍 this is needed, I checked with data dumped from parity and only with this change recovered addresses matches original ones